bin/utils/utils.py

```diff
@@ -0,0 +1,350 @@
+    project_name = get_project_name(final_image_name)
+    result = run(["docker", "ps", "--format={{.Names}} {{.Ports}}"], capture_output=True)
+    output_lines = result.stdout.decode("utf-8").splitlines()
+
+    containers = {}
+    port = None
+
+    for line in output_lines:
+        parts = line.split(None, 1)
+        container_name, ports = parts
+        if container_name.startswith(project_name + "_"):
+            service = container_name[len(project_name) + 1 :].split("_", 1)[0]
+            containers[service] = container_name
+            if service == "gateway":
+                port = get_port_from_docker_ports(ports)
+
+    if containers and port:
+        return {"state": "running", "containers": containers, "port": port}
+    else:
+        return {"state": "stopped"}
+
+
+def get_port_from_docker_ports(info):
+    port_re = r"0\.0\.0\.0:(\d+)->"
+    match = re.match(port_re, info)
+    port = int(match.group(1)) if match else None
+    return port
+
+
+DHIS2_DATA_IMAGE = "dhis2-data"
+
+
+def get_project_name(image_name):
+    """
+    Return the project name prefix from an image name.
+
+    Example: eyeseetea/dhis2-data:2.30-ento -> eyeseetea-2-30-ento
+
+    The final containers created have names: PROJECT_NAME_{SERVICE}_1.
+    """
+    clean_image_name = image_name.replace("/" + DHIS2_DATA_IMAGE, "")
+    name_with_prefix = "{}-{}".format(PROJECT_NAME_PREFIX, clean_image_name)
+    return re.sub(r"[^\w]", "", name_with_prefix)
+
+
+def get_core_image_name(data_image_name):
+    """Return core image name from the data image.
+
+    Example: eyeseetea/dhis2-data:2.30-sierra -> eyeseetea/dhis2-core:2.30
+    """
+    sp1 = data_image_name.split("/", 1)
+    if len(sp1) != 2:
+        raise D2DockerError("Invalid data image name: {}".format(data_image_name))
+    namespace, name_with_tag = sp1
+    sp2 = name_with_tag.split(":", 1)
+    if len(sp2) != 2:
+        raise D2DockerError("Invalid data image name: {}".format(data_image_name))
+    name, tag = sp2
+    sp3 = tag.split("-", 1)
+
+    return "{}/dhis2-core:{}".format(namespace, sp3[0])
+
+
+def run_docker_compose(
+    args,
+    data_image=None,
+    core_image=None,
+    port=None,
+    load_from_data=True,
+    post_sql_dir=None,
+    scripts_dir=None,
+    **kwargs,
+):
+    """
+    Run a docker-compose command for a given image.
+
+    If empty, check if there is one empty, and which port it's running listening to.
+
+    The DHIS2_CORE_IMAGE is inferred from the data repo, if not specified.
+    """
+    final_image_name = data_image or get_running_image_name()
+    project_name = get_project_name(final_image_name)
+    core_image_name = core_image or get_core_image_name(data_image)
+
+    check_directory_for_docker_volume(post_sql_dir)
+    check_directory_for_docker_volume(scripts_dir)
+
+    env_pairs = [
+        ("DHIS2_DATA_IMAGE", final_image_name),
+        ("DHIS2_CORE_PORT", str(port)) if port else None,
+        ("DHIS2_CORE_CONTEXT_PATH", ""),
```

Comment:
This appears to be unused now, do we want to pass a context path to nginx?

We need to be careful here as this could potentially break in other OS environments

Reply:
Yes, it's now unused. I don't think we'll need contexts for now, I will remove it.